### PR TITLE
glibc: implement getopt_long_only

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Comment by no92** [comment]
Ready for review now.

File: options/glibc/generic/getopt-stubs.cpp [hunk 1]
```diff
@@ -1,36 +1,87 @@
...
 
+#include <frg/optional.hpp>
 #include <mlibc/debug.hpp>
 
 char *optarg;
 int optind = 1;
 int opterr = 1;
 int optopt;
+int __optreset = 0;
+extern int optreset __attribute__((weak, alias("__optreset")));
```
Comment (64) [inline options/glibc/generic/getopt-stubs.cpp:16]: ifdef this behind BSD option?

File: tests/glibc/getopt.c [hunk 2]
```diff
@@ -235,6 +236,46 @@ void test7() {
...
+	int foo = false;
+	int bar = false;
+
+	const struct option longopts[] = {
+		{"foo", required_argument, &foo, 'x'},
+		{"bar", required_argument, &bar, 'y'},
+		{NULL, no_argument, NULL, 0}
+	};
+
+	int test_argc = 7;
```
Comment (64) [inline tests/glibc/getopt.c:251]: 6?

**Review by 64 — CHANGES_REQUESTED** [review]

File: options/glibc/generic/getopt-stubs.cpp [hunk 3]
```diff
@@ -15,22 +14,78 @@ int opterr = 1;
...
 
 namespace {
-	int __optpos = 1;
-}
 
-int getopt_long(int argc, char * const argv[], const char *optstring,
-		const struct option *longopts, int *longindex) {
+int __optpos = 1;
+
+enum GetoptMode {
```
Comment (64) [inline options/glibc/generic/getopt-stubs.cpp:20]: enum class?

**Review by 64 — APPROVED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**